src/ServiceStack.OrmLite.SqlServer/SqlServerOrmLiteDialectProvider.cs

```diff
@@ -155,7 +153,7 @@ public override string GetForeignKeyOnDeleteClause(ForeignKeyConstraint foreignK
 
         public override string GetForeignKeyOnUpdateClause(ForeignKeyConstraint foreignKey)
         {
-            return "RESTRICT" == (foreignKey.OnDelete ?? "").ToUpper()
+            return "RESTRICT" == (foreignKey.OnUpdate ?? "").ToUpper()
```

Comment:
Looks like GetForeignKeyOnUpdateClause is a copy of GetForeignKeyOnDeleteClause, but was not updated.